plugins/utils/nvim-ufo.nix

```diff
@@ -0,0 +1,80 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.nvim-ufo;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.nvim-ufo =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "nvim-ufo";
+
+      package = helpers.mkPackageOption "nvim-ufo" pkgs.vimPlugins.nvim-ufo;
+
+      openFoldHighlightTimeout = helpers.defaultNullOpts.mkInt 400 ''
```

Comment:
For the option names we try to map very closely to the upstream names, so I think `openFoldHlTimeout` would be more appropriate